packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/CellValue.js

```diff
@@ -27,13 +27,31 @@ const CellValue = ({ type, value }) => {
     });
   }
 
-  if (['float', 'integer', 'biginteger', 'decimal'].includes(type)) {
-    formattedValue = formatNumber(value);
+  if (['float', 'decimal'].includes(type)) {
+    const numberOfDecimals = getNumberOfDecimals(value);
+
+    formattedValue = formatNumber(value, {
+      minimumFractionDigits: numberOfDecimals,
+      maximumFractionDigits: numberOfDecimals,
+    });
+  }
+
+  if (['integer', 'biginteger'].includes(type)) {
+    formattedValue = formatNumber(value, { maximumFractionDigits: 0 });
   }
 
   return toString(formattedValue);
 };
 
+const getNumberOfDecimals = value => {
+  if (value % 1 !== 0) {
+    // value has decimals
+    return value.toString().split('.')[1].length;
```

Comment:
Unfortunately this doesn't apply to every language. In german e.g. the decimal seperator is ",". The design-system has already solved this by using `@internationalized/number`.

https://github.com/strapi/design-system/blob/develop/packages/strapi-design-system/src/NumberInput/NumberInput.js#L6

Would it make sense to unify this behavoir?

Reply:
I see that I first misunderstood the bug. This PR solves the problem of values rounding in the table in content manager, but not the click outside in input field